lib/pco_api/record_association.ex

```diff
@@ -4,16 +4,19 @@ defmodule PcoApi.RecordAssociation do
       def unquote(:"#{name}")([]), do: []
       def unquote(:"#{name}")(%PcoApi.Record{} = record) do
         # TODO: error handling when the association link doesn't exist
-        request(:get, record.links[Atom.to_string(unquote(name))], [])
-        |> Enum.map(fn(%{"attributes" => attrs, "id" => id, "links" => links, "type" => type}) ->
-          %PcoApi.Record{attributes: attrs, id: id, links: links, type: type}
-        end)
+        unquote(:"#{name}")(request(:get, record.links[Atom.to_string(unquote(name))], []))
       end
       def unquote(:"#{name}")([%PcoApi.Record{} = first | rest]) do
         [unquote(:"#{name}")(first) | unquote(:"#{name}")(rest)]
         |> List.flatten
         |> Enum.dedup
       end
+      def unquote(:"#{name}")(%{"attributes" => attrs, "id" => id, "links" => links, "type" => type}) do
+        %PcoApi.Record{attributes: attrs, id: id, links: links, type: type}
+      end
+      def unquote(:"#{name}")(list) when is_list(list) do
+        list |> Enum.map(&(unquote(:"#{name}")(&1)))
+      end
```

Comment:
The line 5 `get` was expecting `request/3` to return a list which it would iterate through and map each item inside. Some requests return a single record and the Enum.map would fail. So I tore out the "assigning `%<"attributes"...>` to a record" logic and made it it's own function (line 14) and then moved the Enum.map part of the function out so that it only runs when a list is returned, otherwise it goes straight to assign the attributes to a record.